packages/gatsby/src/redux/run-sift.js

```diff
@@ -181,13 +182,27 @@ const getBucketsForFilters = (filters, nodeTypeNames, filtersCache) => {
   // Fail fast while trying to create and get the value-cache for each path
   let every = filters.every((filter /*: DbQuery*/) => {
     let cacheKey = createTypedFilterCacheKey(nodeTypeNames, filter)
-    getBucketsForQueryFilter(
-      cacheKey,
-      filter,
-      nodeTypeNames,
-      filtersCache,
-      filterCaches
-    )
+    if (filter.type === `query`) {
+      // (Let TS warn us if a new query type gets added)
+      const q /*: IDbQueryQuery */ = filter
+      return getBucketsForQueryFilter(
```

Comment:
Because one (root) query may have multiple filters, and each filter may return multiple caches. (Potentially even the same ones, although that would be a bug in the site.) So I prefer to do the squash only once, when all filters have been processed.